### PR TITLE
Added changes to avoid setting empty style for GeoJson object 

### PR DESCRIPTION
**Pull request by omanges**
This is a fix for the bug reported in https://github.com/jupyter-widgets/ipyleaflet/issues/555

### CODE REVIEW

**Comment by martinRenou** [comment]
Thanks a lot for opening a PR. I am not forgetting about this, we need to merge it before making a release.

File: ipyleaflet/leaflet.py [hunk 1]
```diff
@@ -579,7 +579,7 @@ def _get_data(self):
             elif self.data['type'] == 'FeatureCollection':
                 for feature in self.data['features']:
                     feature['properties']['style'] = self.style_callback(feature)
-        else:
+        elif self.style:
             if self.data['type'] == 'Feature':
                 self.data['properties']['style'] = self.style
```
Comment (martinRenou) [inline ipyleaflet/leaflet.py:584]: In the method, we should replace all the `self.data['properties']['style'] = something` into `self.data['properties']['style'].update(something)`

Comment (martinRenou) [reply ipyleaflet/leaflet.py:584]: this way we never overwrite the style, we simply update it.

Comment (omanges) [reply ipyleaflet/leaflet.py:580]: But if the style key is not present within the `feature` then will it give a `KeyError`

Comment (omanges) [reply ipyleaflet/leaflet.py]: Made changes as suggested :)

File: ipyleaflet/leaflet.py [hunk 2]
```diff
@@ -571,16 +571,16 @@ def _update_data(self, change):
...
-        else:
+                    feature['properties']['style'].update(self.style_callback(feature))
+        elif self.style:
             if self.data['type'] == 'Feature':
-                self.data['properties']['style'] = self.style
+                self.data['properties']['style'].update(self.style)
             elif self.data['type'] == 'FeatureCollection':
                 for feature in self.data['features']:
-                    feature['properties']['style'] = self.style
+                    feature['properties']['style'].update(self.style)
```
Comment (sackh) [inline ipyleaflet/leaflet.py:583]: There should be a minor change here, as this code always expects that there will be `style` in properties of `geojson`. If `style` is not present in the properties this fails with `KeyError` exception. I think there should be one more check if `style` key is already present then update, else assign the value.
e.g. - 
```python
if 'style' in  feature['properties']:
   feature['properties']['style'].update(self.style)
else:
   feature['properties']['style'] = self.style
```

Comment (sackh) [reply ipyleaflet/leaflet.py:583]: for all update statements, need to check if `style` key is already present or not.

Comment (sackh) [reply ipyleaflet/leaflet.py:583]: I have raised PR #572

**Pull request merged**